src/math/ceil.rs

```diff
@@ -12,7 +12,7 @@ pub fn ceil(x: f64) -> f64 {
     // `f64.ceil` native instruction, so we can leverage this for both code size
     // and speed.
     llvm_intrinsically_optimized! {
-        #[cfg(target_arch = "wasm32")] {
+        #[cfg(all(target_arch = "wasm32", not(feature = "only-soft-floats")))] {
```

Comment:
Should this be done in the `llvm_intrinsically_optimized!` macro definition instead?

Reply:
Seems like this `cfg` (`target_arch = wasm32`) is always used in this macro call, so might as well move it directly into it, good idea!

Reply:
Oh, I meant the `not(feature = "only-soft-floats")` bit, but I don't have strong feelings either way.

Reply:
That's actually what I meant too. ^^'